code/modules/mob/living/carbon/emote.dm

```diff
@@ -87,31 +87,29 @@
 	key = "meow"
 	key_third_person = "meows"
 	vary = TRUE
+	sound = SFX_CAT_MEOW
+	message = "meows!"
+	message_mime = "meows silently."
+	emote_type = EMOTE_VISIBLE | EMOTE_AUDIBLE
 
-/datum/emote/living/carbon/meow/run_emote(mob/user, params, type_override, intentional)
-	. = ..()
-	if(istype(user.get_organ_slot(ORGAN_SLOT_TONGUE), /obj/item/organ/tongue/cat))
-		sound = SFX_CAT_MEOW
-		message = "meows!"
-		emote_type = EMOTE_VISIBLE | EMOTE_AUDIBLE
-	else
-		to_chat(user, span_warning("You can't quite make the sound."))
-		return
+/datum/emote/living/carbon/meow/can_run_emote(mob/living/carbon/user, status_check = TRUE , intentional, params)
+	if(!iscarbon(user) || (!istype(user.get_organ_slot(ORGAN_SLOT_TONGUE), /obj/item/organ/tongue/cat)))
+		return FALSE
+	return ..()
 
 /datum/emote/living/carbon/purr
 	key = "purr"
 	key_third_person = "purrs"
 	vary = TRUE
+	sound = SFX_CAT_PURR
+	message = "purrs."
+	message_mime = "purrs silently."
```

Comment:
Not sure how you'd purr silently, maybe pantomime instead?

Reply:
I'll try and think of something.

Reply:
Maybe "imitates a tongue trill."?

Reply:
interesting problem cause if you do that then it'd make sense to be visible for mimes, so shrug

Reply:
Maybe I could just remove the mime's capability.

Reply:
Tried a fix for vow of silence users not being able to purr, looks like it's working on my test server.